Makefile

```
@@ -0,0 +1,9 @@
+VENV := . venv/bin/activate
+
+types:
+	${VENV} && mypy *.py
+	
+lint:
+	${VENV} && ruff check
+
+static_checking: lint types
```

Optionally simplify this as it feels awkward to repeatedly enable the virtenv when we don't actually need it (made change locally and confirmed this works).  

optional!

```suggestion
types:
        ./venv/bin/mypy *.py

lint:
        ./venv/bin/ruff check

static_checking: lint types
```